src/Microsoft.Health.Fhir.Api/Features/Bundle/BundleAwareJwtBearerHandler.cs

```diff
@@ -0,0 +1,44 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Health.Fhir.Api.Features.Bundle
+{
+    public class BundleAwareJwtBearerHandler : JwtBearerHandler
+    {
+        private readonly IBundleHttpContextAccessor _bundleHttpContextAccessor;
+
+        public BundleAwareJwtBearerHandler(IOptionsMonitor<JwtBearerOptions> options, ILoggerFactory logger, UrlEncoder encoder, IDataProtectionProvider dataProtection, ISystemClock clock, IBundleHttpContextAccessor bundleHttpContextAccessor)
+            : base(options, logger, encoder, dataProtection, clock)
+        {
+            EnsureArg.IsNotNull(bundleHttpContextAccessor, nameof(bundleHttpContextAccessor));
+
+            _bundleHttpContextAccessor = bundleHttpContextAccessor;
+        }
+
+        /// <summary>
+        /// Override to check if the forbidden request is part of a bundle (batch/transaction). If it is, then set the status code of the internal request.
+        /// </summary>
+        /// <param name="properties">The authentication properties</param>
+        /// <returns>Returns internal HandleForbiddenAsync Task.</returns>
+        protected override async Task HandleForbiddenAsync(AuthenticationProperties properties)
```

Comment:
This is set up as part of the authentication pipeline in MVC. When a route is invoked, this method is called when it's determined that there is no authorization for the given route. 

Because the pipeline is only setup once, the httpContext is the main httpContext on the request (`POST` on `/` in the case of bundles/transactions) and the httpContext doesn't correctly flow back into the bundler handler. 

To counteract this, we have an httpContextAccessor that only has a value if it's part of a bundle/transaction that is set to the sub context thats created inside the `BundleHandler`.

Reply:
I'd put this explanation on the as a comment on the class